src/main.rs

```diff
@@ -431,49 +542,82 @@ fn main() -> Result<()> {
         }
 
         if let Some(output) = args.output {
-            if ext_normal(&output) != ext_normal(&args.path) {
-                bail!("Output type must match source type");
-            }
-            if output.exists() && !args.force {
-                bail!("Output already exists, use -f/force to force write");
-            }
-
-            if output.file_name().is_none() {
-                bail!("Missing filename on output");
-            }
-            if output.extension().is_none() {
-                bail!("Missing extension output");
-            }
-
-            let signer = if let Some(signer_process_name) = args.signer_path {
-                let cb_config = CallbackSignerConfig::new(&sign_config, args.reserve_size)?;
-
-                let process_runner = Box::new(ExternalProcessRunner::new(
-                    cb_config.clone(),
-                    signer_process_name,
-                ));
-                let signer = CallbackSigner::new(process_runner, cb_config);
+            // fragmented embedding
+            if let Some(Commands::Fragment { fragments_glob }) = &args.command {
+                if output.exists() && !output.is_dir() {
+                    bail!("Output cannot point to existing file, must be a directory");
+                }
 
-                Box::new(signer)
+                let signer = if let Some(signer_process_name) = args.signer_path {
```

Comment:
move this above the if let Some(Commands:Fragment) test so it is shared by anything with an output.